Sources/KlaviyoLocation/KlaviyoLocationManager.swift

```diff
@@ -107,4 +113,49 @@ class KlaviyoLocationManager: NSObject {
 
         regions.forEach(locationManager.stopMonitoring)
     }
+
+    // MARK: - API Key Observation
+
+    @MainActor
+    private func startObservingAPIKeyChanges() {
+        guard apiKeyCancellable == nil else { return }
+        apiKeyCancellable = KlaviyoInternal.apiKeyPublisher()
+            .receive(on: DispatchQueue.main)
+            .removeDuplicates()
+            .sink { [weak self] result in
+                guard let self else { return }
+                switch result {
+                case let .success(apiKey):
+                    if #available(iOS 14.0, *) {
+                        Logger.geoservices.info("🔄 Company ID changed. Updating geofences for new company: \(apiKey)")
+                    }
+                    startGeofenceMonitoring()
+                case .failure:
+                    break
+                }
+            }
+    }
+
+    // MARK: - Lifecycle Observation
+
+    @MainActor
+    private func startObservingLifecycleChanges() {
```

Comment:
Is this to detect permission change?
And same as above, does this need to be "stopped" on unregisterGeofencing?

Reply:
A bit nitpicky, but I think this will result in logging `"App does not have 'authorizedAlways' permission to access the user's location"` every single app foreground if user hasn't given permission which just seems pretty noisey.